react/PlaceOrder.tsx

```diff
@@ -1,14 +1,138 @@
-import React from 'react'
+import React, { useState } from 'react'
 import { FormattedMessage } from 'react-intl'
 import { Button } from 'vtex.styleguide'
+import { OrderForm } from 'vtex.order-manager'
+import postRobot from 'post-robot'
+import { useRuntime } from 'vtex.render-runtime'
+import { OrderPayment } from 'vtex.order-payment'
 
-import styles from './PlaceOrder.css'
+const { useOrderForm } = OrderForm
+const { useOrderPayment } = OrderPayment
 
 const PlaceOrder: React.FC = () => {
+  const { orderForm } = useOrderForm()
+  const { value, referenceValue, interestValue } = useOrderPayment()
+  const {
+    culture: { currency },
+    rootPath = '',
+  } = useRuntime()
+
+  const [placingOrder, setPlacingOrder] = useState(false)
+  const [cardFormIframe] = useState(
+    () => document.getElementById('chk-card-form')! as HTMLIFrameElement
```

Comment:
What is this `HTMLIFrameElement`?

Reply:
it's the typescript interface for the instance of an iframe, so we can get autocomplete on stuff like `cardFormIframe.contentWindow.etc`

Reply:
Nice.